tools/adventure-pack/goodies/typescript/Iterator.prototype.every/index.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "./index";
+
+describe("Iterator.prototype.every", () => {
+  it("returns true if all elements pass the test and false if an element fails a test", () => {
+    const isEven = (element: number) => element % 2 === 0;
+    expect([2, 4, 6, 8, 10].values().every(isEven)).toBe(true);
+    expect([2, 4, 6, 7, 8, 10].values().every(isEven)).toBe(false);
+    expect([1, 4, 6, 8, 10].values().every(isEven)).toBe(false);
+    expect([2, 4, 6, 8, 11].values().every(isEven)).toBe(false);
+  });
+
+  it("returns true for an empty iterator", () => {
+    expect([].values().every((element) => element > 5)).toBe(true);
+  });
+
+  it("can test Map entries()", () => {
+    const map = new Map([
+      ["H", 1],
+      ["e", -2],
+      ["L", 3],
+      ["l", 0],
+      ["O", 5],
+      ["W", -1],
+      ["O", 0],
+      ["r", -5],
+      ["L", 1],
+      ["D", 2],
+    ]);
+    const callbackFn = ([key, value]: [string, number]) =>
+      key === key.toUpperCase() && value > 0;
```

Comment:
If you inline callbackFn you probably won't need to specify the type, TypeScript will have enough context to infer it.

Reply:
TS still gives me type issues 

`Binding element 'key' implicitly has an 'any' type.ts(7031)
(parameter) key: any` 

and
 
`Binding element 'value' implicitly has an 'any' type.ts(7031)
(parameter) value: any`